pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/ReadableSequentialData.java

```diff
@@ -214,7 +214,10 @@ default long readBytes(@NonNull final BufferedData dst) throws UncheckedIOExcept
         }
 
         final var bytes = new byte[length];
-        readBytes(bytes, 0, length);
+        final long bytesRead = readBytes(bytes, 0, length);
+        if (bytesRead != length) {
+            throw new BufferUnderflowException();
```

Comment:
Minor: might want to provide some context in the exception message, like the number of expected bytes to read and the actual number read. That could help debugging this exception in the future.

Reply:
`BufferUnderflowException` does not accept any arguments unfortunately. In order to add a message, we would have to change the exception type, but this would be a big change, since all the javadocs declare that this exception is thrown.